src/main/java/org/sonatype/repository/chef/internal/util/CookBookApiAbsoluteUrlRemover.java

```diff
@@ -0,0 +1,236 @@
+      getWriter().name(name);
+      JsonToken peek = getReader().peek();
+      if (peek.equals(JsonToken.STRING) || (name.equals("cookbook"))) {
+        doSetUrlAsRelative(getReader(), getWriter());
+      }
+    }
+  }
+
+  private class CookbookDetailsByVersionJsonStreamer
+      extends JsonStreamer
+  {
+    public CookbookDetailsByVersionJsonStreamer(final JsonReader reader, final JsonWriter writer) {
+      super(reader, writer);
+    }
+
+    @Override
+    public void getAndSetName() throws IOException {
+      String name = getReader().nextName();
+      getWriter().name(name);
+      JsonToken peek = getReader().peek();
+      if (peek.equals(JsonToken.STRING) || (name.equals("cookbook") || name.equals("file"))) {
+        doSetUrlAsRelative(getReader(), getWriter());
+      }
+    }
+  }
+
+  private class CookbookDetailsJsonStreamer
+    extends JsonStreamer
+  {
+    public CookbookDetailsJsonStreamer(final JsonReader reader, final JsonWriter writer) {
+      super(reader, writer);
+    }
+
+    @Override
+    public void getAndSetName() throws IOException {
+      String name = getReader().nextName();
+      getWriter().name(name);
+      JsonToken peek = getReader().peek();
+      if (peek.equals(JsonToken.STRING) || (name.equals("versions") && peek.equals(JsonToken.BEGIN_ARRAY))) {
+        maybeSetUrlToRelativeCase(getReader(), getWriter(), name);
+      }
+    }
+  }
+
+  private class CookbookListJsonStreamer
+    extends JsonStreamer
+  {
+    public CookbookListJsonStreamer(final JsonReader reader, final JsonWriter writer) {
+      super(reader, writer);
+    }
+
+    @Override
+    public void getAndSetName() throws IOException {
+      String name = getReader().nextName();
+      getWriter().name(name);
+      JsonToken peek = getReader().peek();
+      if (peek.equals(JsonToken.STRING) && "cookbook".equals(name)) {
+        maybeSetUrlToRelative(getReader(), getWriter());
+      }
+    }
+  }
+
+  private void maybeSetUrlToRelativeCase(final JsonReader reader, final JsonWriter writer, final String name)
+      throws IOException
+  {
+    switch (name) {
+      case "latest_version":
+        doSetUrlAsRelative(reader, writer);
+        break;
+      case "versions":
+        reader.beginArray();
+        writer.beginArray();
+        while (reader.hasNext()) {
+          doSetUrlAsRelative(reader, writer);
+        }
+        reader.endArray();
+        writer.endArray();
+        break;
+      default:
+        writer.value(reader.nextString());
+        break;
+    }
+  }
+
+  private void doSetUrlAsRelative(final JsonReader reader, final JsonWriter writer)
+      throws IOException
+  {
+    String url = reader.nextString();
+    try {
+      URI uri = new URIBuilder(url).build();
+      if (uri.isAbsolute()) {
+        String rightHand = uri.getPath();
+
+        writer.value(rightHand);
+      }
+      else {
+        writer.value(url);
+      }
+    } catch (URISyntaxException ex) {
```

Comment:
Question: Under what circumstances can this happen, should we be logging something even if it is at debug level?